lib/modules/dosomething/dosomething_api/resources/reportback_item_resource.inc

```diff
@@ -133,9 +141,19 @@ function _reportback_item_resource_access($op) {
  * @param  bool    $random
  * @param  int     $page
  * @param  bool    $load_user
+ * @param  int     $reaction_user
  * @return array
  */
-function _reportback_item_resource_index($campaigns, $exclude, $status, $count, $random, $page, $load_user) {
+function _reportback_item_resource_index($campaigns, $exclude, $status, $count, $random, $page, $load_user, $reaction_user) {
+  if ($reaction_user) {
+    $reaction_user = user_load($reaction_user);
+
+    if ($reaction_user) {
+      global $user;
+      $user = $reaction_user;
+    }
+  }
+
```

Comment:
The above logic shouldn't happen in the resource file. It's supposed to just pass the params into the `$parameters` array and then this logic should happen elsewhere... Maybe in the transformer as a separate function called in the `index()` method.